src/VerifPal/Pretty.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE MultiParamTypeClasses #-}
+module VerifPal.Pretty where
+
+import Data.Function ((&))
+import Data.Text (Text)
+import qualified Data.Text as Text
+import Data.Void
+import Error.Diagnose.Compat.Megaparsec (HasHints (hints))
+import Prettyprinter (Doc, Pretty (..), align, annotate, colon, hardline, lbracket, rbracket, space, width, (<+>))
+import Prettyprinter.Render.Terminal (AnsiStyle, Color (..), bold, color, colorDull, putDoc)
+import Text.Show.Pretty (ppDoc)
+import VerifPal.Types (Constant (..), Query (..), QueryKind (..))
+import VerifPal.Check (ModelState,CanonExpr)
+import Data.Graph.Inductive (Gr,OrdGr)
+
+prettifyQuery :: QueryKind -> String
+prettifyQuery (FreshnessQuery const) =
+  "freshness? " ++ Text.unpack (constantName const)
+prettifyQuery (EquivalenceQuery consts) =
+  "equivalence? "
+    ++ ( map constantName consts
+           & Text.intercalate ", "
+           & Text.unpack
+       )
+prettifyQuery k = show k
+
+-- see: https://github.com/Mesabloo/diagnose/blob/master/test/rendering/Spec.hs
+instance HasHints Void Text where
+  -- hints :: Void -> [Text]
+  hints _ = ["TODOwhatisthis"]
```

Comment:
I'm not sure if this type class instance is required by something?

Reply:
It was required by the `diagnose` parser library; or at least it was complaining without this. I don't really understand *why*, but it didn't seem to build without it.